frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -0,0 +1,251 @@
+import d3 from "d3";
+import _ from "underscore";
+import type {
+  DataKey,
+  Extent,
+  SeriesExtents,
+  SeriesModel,
+} from "metabase/visualizations/echarts/cartesian/model/types";
+import type { ComputedVisualizationSettings } from "metabase/visualizations/types";
+import type { SeriesSettings } from "metabase-types/api";
+import { isNotNull } from "metabase/lib/types";
+
+const KEYS_TO_COMPARE = new Set([
+  "number_style",
+  "currency",
+  "currency_style",
+  "number_separators",
+  "decimals",
+  "scale",
+  "prefix",
+  "suffix",
+]);
+
+export function getLineAreaBarComparisonSettings(
+  columnSettings: Record<string, unknown>,
+) {
+  return _.pick(columnSettings, (value, key) => {
+    if (!KEYS_TO_COMPARE.has(key)) {
+      return false;
+    }
+    if ((key === "prefix" || key === "suffix") && value === "") {
+      return false;
+    }
+    return true;
+  });
+}
+
+const getAggregateColumnsCount = (seriesModels: SeriesModel[]) => {
+  return _.uniq(
+    seriesModels
+      .map(seriesModel => seriesModel.column)
+      .filter(column => column.source === "aggregation")
+      .map(column => column.name),
+  ).length;
+};
+
+export function shouldAutoSplitYAxis(
+  settings: ComputedVisualizationSettings,
+  seriesModels: SeriesModel[],
+  seriesExtents: SeriesExtents,
+) {
+  const hasSingleCard =
+    seriesModels.reduce((cardIds, seriesModel) => {
+      cardIds.add(seriesModel.cardId);
+      return cardIds;
+    }, new Set<number | undefined>()).size === 1;
+
+  if (
+    settings["graph.y_axis.auto_split"] === false ||
+    (hasSingleCard && getAggregateColumnsCount(seriesModels) < 2) ||
```

Comment:
Our logic of using the number of aggregate columns to count how many metric columns there are was flawed, because it didn't work for SQL queries. On SQL queries `column.source` is always `"native"`. 

See this thread on slack: https://metaboat.slack.com/archives/C01LQQ2UW03/p1699633983871819?thread_ts=1699568456.519179&cid=C01LQQ2UW03